README.md

```diff
@@ -758,6 +759,49 @@ Extra note, this is serialization benchmark result.
 
  Of course, `IntKey` is fastest but `StringKey` also performs reasonably well.
 
+### <a name="string-interning"></a>String interning
+
+The msgpack format does not provide for reusing strings in the data stream.
+This naturally leads the deserializer to create a new `string` object for every string encountered,
+even if it is equal to another string previously encountered.
+
+When deserializing data that may contain the same strings repeatedly it can be worthwhile
+to have the deserializer take a little extra time to check whether it has seen a given string before
+and reuse it if it has.
+
+To enable string interning on *all* string values, use a resolver that specifies `StringInterningFormatter`
+before any of the standard ones, like this:
+
+```cs
+var options = MessagePackSerializerOptions.Standard.WithResolver(
```

Comment:
By being optional, there is no problem in provide it.
However, I think that Resolver should also be provided as a counterpart to Formatter.
That is, a `StringInterningResolver`.

Reply:
Would this resolver *only* point to the one formatter, such that it would still need to be combined with other resolvers for every case?

I'm going to complete this PR for now since we can always add a resolver in a subsequent PR based on your answer.